staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -404,7 +404,15 @@ type eventBroadcasterAdapterImpl struct {
 
 // NewEventBroadcasterAdapter creates a wrapper around new and legacy broadcasters to simplify
 // migration of individual components to the new Event API.
-func NewEventBroadcasterAdapter(client clientset.Interface) EventBroadcasterAdapter {
```

Comment:
changing the public signature forces the consumers to have to do manual job on revendorings, can we keep the existing one, that is also how golang made this changes, Dial --> DialWithContext